src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -121,6 +121,28 @@ export class InstallAction extends Action {
 	}
 
 	private install(extension: IExtension): TPromise<void> {
+		// {{SQL CARBON EDIT}}
+		if (this.extension.publisher !== 'Microsoft') {
+			this.windowService.showMessageBox({
+				type: 'warning',
+				buttons: [localize('thirdPartExt.yes', 'Yes'), localize('thirdPartyExt.no', 'No')],
+				message: localize('thirdPartyExtension', 'Disclaimer: "{0}" is a third party extension and might involve security risks involved. Are you sure you want to install this extension?', this.extension.displayName)
```

Comment:
Message has an extra "involved". We may want to run this by PMs for the exact wording but otherwise it looks fine, though I don't think we need "Disclaimer" at the start either

Reply:
Good catch